DOCSP-29541: newclient and connect deprecation notice (#258)
mongodb/docs-golang: source/whats-new.txt

* :ref:`Version 1.1 <version-1.1>`
* :ref:`Version 1.0 <version-1.0>`

> .. _version-1.12.0:

Reviewer: Q: I see that we have this for v11, but do we need it? I've only seen ".0" patch versions added if/when a ".1" patch is released to distinguish the patch versions.

.. _version-1.12:

What's New in 1.12
------------------

.. important:: Deprecation Notice

>    - The ``mongo.NewClient()`` and ``client.Connect()`` methods are
>      deprecated because you can create a client and connect in one call
>      by using the ``mongo.Connect()`` method.

Reviewer: S: Sentence is maybe a little long

```suggestion
   - The ``mongo.NewClient()`` and ``client.Connect()`` methods are
     deprecated. Use the ``mongo.Connect()`` method to create a client 
     and connect in one call. 
```
